### DISCUSSION TITLE
Project roadmap - discussion

### DISCUSSION THREAD

**Discussion by sukesh-ak** [sukesh-ak/ESP32-TUX#10] [General]
Planning to port ESP32-TUX to more platforms soon

- [ ] RPI-TUX => Raspberry Pi OS & other Linux platforms
- [ ] RPI-TUX => Raspberry Pi - Circle - Bare-metal
- [ ] WEB-TUX => Web - Emscripten 
- [ ] ESP32-TUX-Mini - For smaller screen devices with capacitive touch - different navigation styles.
- [ ] Sipeed (https://www.indiegogo.com/projects/sipeed-maix-new-experience-to-risc-v-aiot-tinyml)

What do you guys think?

**Comment by modi12jin** [comment]
@sukesh-ak Allwinner D1s or Allwinner T113-S3 are good choices, here is its running LVGL video

【搭载全志t113的麻雀mq-r跑lvgl 100帧-哔哩哔哩】 https://b23.tv/sWqFGcB

【T113-S3开发板视频解码、LVGL、Qt测试-哔哩哔哩】 https://b23.tv/Rdb7Bjc

【全志科技 T113-S3 驱动 720*1440 MIPI 电容触摸屏-哔哩哔哩】 https://b23.tv/psqJ1bM

【LVGL，畅享丝滑~-哔哩哔哩】 https://b23.tv/IPbbRHE

You can buy chips or development boards, and the information is relatively open, but you need to have some experience

**Reply by sukesh-ak** [reply]
Great collection. I have LVGL working on Linux and Web already. Will need to port the remaining code and clean up the task related things so that everything works using `lv_msg` based UI API.

**Reply by modi12jin** [reply]
![tb_image_share_1671188204701](https://user-images.githubusercontent.com/40233017/208083686-fc5d3596-d034-4cd1-8e7e-44acd9e735e5.jpg)

**Reply by modi12jin** [reply]
https://mobile.twitter.com/mangopi_sbc

**Reply by sukesh-ak** [reply]
They all have good stuff but not available here to play with it. Maybe I should ask them for samples :)

**Comment by modi12jin** [comment]
Sipeed is an exclusive supply for this board chip, so it will be difficult for you to build a PCB from the chip level

**Reply by sukesh-ak** [reply]
I see they do have a module/SOM? 
https://www.indiegogo.com/projects/sipeed-maix-new-experience-to-risc-v-aiot-tinyml/

**Reply by modi12jin** [reply]
If it is a module, you can give it a try. I don’t know if the openness of the data is enough to meet the development needs.

**Reply by sukesh-ak** [reply]
I saw they support Linux so I was cool with it :)

**Reply by modi12jin** [reply]
![tb_image_share_1671188711439](https://user-images.githubusercontent.com/40233017/208085033-ea11a9cb-126e-4e39-990b-b97edf538f55.jpg)

 The module is almost the same as the esp32 module and also has WiFi

Seems like the screen interface only supports SPI?

**Reply by modi12jin** [reply]
@sukesh-ak I know that this 1.69-inch screen is purchased separately

Official website: www.polcd.com 
Alibaba international station store: https://polcd.en.alibaba.com/

**Reply by sukesh-ak** [reply]
> @sukesh-ak I know that this 1.69-inch screen is purchased separately
> 
> Official website: [www.polcd.com](http://www.polcd.com) Alibaba international station store: https://polcd.en.alibaba.com/

Will add to my display supplier collection. Other 2 I know are
http://en.tft-tft.com/

https://www.buydisplay.com/

**Comment by modi12jin** [comment]
@sukesh-ak I saw someone build LVGL with Android

**Reply by sukesh-ak** [reply]
Yes you can just change the driver to port easily.
I have something similar using ImGui
https://github.com/sukesh-ak/ImGui-GLFW-OpenGL3

**Reply by modi12jin** [reply]
https://blog.csdn.net/jayyuz/article/details/123842040

**Comment by modi12jin** [comment]
@sukesh-ak SWM34 supports hardware decoding jpeg

【让二次元小姐姐在LVGL界面上唱歌跳舞（SWM34SRE）-哔哩哔哩】 https://b23.tv/6S4gzER

【国产单片机播放高清视频（SWM34SRE，800x480）-哔哩哔哩】 https://b23.tv/JNtJHid

**Reply by sukesh-ak** [reply]
> @sukesh-ak SWM34 supports hardware decoding jpeg
> 
> 【让二次元小姐姐在LVGL界面上唱歌跳舞（SWM34SRE）-哔哩哔哩】 https://b23.tv/6S4gzER
> 
> 【国产单片机播放高清视频（SWM34SRE，800x480）-哔哩哔哩】 https://b23.tv/JNtJHid

I need these boards and another 24hrs in a day 💯

**Comment by modi12jin** [comment]
https://www.cnx-software.com/2022/12/02/pine64-ox64-sbc-bl808-risc-v-multi-protocol-wisoc-64mb-ram/

**Reply by modi12jin** [reply]
@sukesh-ak unbelievable

**Reply by sukesh-ak** [reply]
Need to get one but I need Linux on it.

**Reply by modi12jin** [reply]
BL808D supports mipi DSI, BL808C does not support mipi DSI

**Reply by sukesh-ak** [reply]
Is this using BL808D?

**Reply by sukesh-ak** [reply]
It is 😬

**Reply by sukesh-ak** [reply]
https://twitter.com/sipeedio/status/1606246191749947392?s=46&t=vCCRsYKKAnYWAgG_MgW0GQ

**Reply by modi12jin** [reply]
> https://twitter.com/sipeedio/status/1606246191749947392?s=46&t=vCCRsYKKAnYWAgG_MgW0GQ

BL618

**Comment by modi12jin** [comment]
@sukesh-ak I believe you should know that ESP32-P4 exists.

 As far as I know, it may be mass-produced in Q4 this year, and there is already a test version of the chip

**Reply by sukesh-ak** [reply]
Thanks for the info. I already saw it but sad its missing Wi-Fi/BLE :(

**Reply by modi12jin** [reply]
The best WIFI partner of ESP32-P4 is C5 or C6

**Reply by sukesh-ak** [reply]
> The best WIFI partner of ESP32-P4 is C5 or C6

Will wait to see the smallest P4 dev board with WiFi/BLE

**Comment by powersoft** [comment]
Hello,
I have an sc01-plus, and it is working fine. The only thing I can't get running is the sd-card.
I use the Arduino envoirment to program. Do you have tips for me to get it running.

Cheers,
Jan Kromhout
Hellevoetsluis-NL

**Reply by sukesh-ak** [reply]
> Hello, I have an sc01-plus, and it is working fine. The only thing I can't get running is the sd-card. I use the Arduino envoirment to program. Do you have tips for me to get it running.
> 
> Cheers, Jan Kromhout Hellevoetsluis-NL

Please ask questions in a different thread when it's not related to this. You can get device config details here
https://github.com/sukesh-ak/ESP32-TUX/blob/master/main/devices/conf_WT32SCO1-Plus.h

**Comment by powersoft** [comment]
Thanks for the response.
If I have the wrong thread please can you give me the right one!

This is my code and I include the config file (from your github page)without any results.
Should I include other stuff to get is working?

Cheers,
Jan


#define LGFX_USE_V1         // set to use new version of library
#include <LovyanGFX.hpp>    // main library
#include "config.h"


#include "FS.h"       // File System header
#include "SD.h"       // SD Card header 


static LGFX lcd;            // declare display variable

#define sdSelectPin 41

void setup() {
  Serial.begin(500000);
  lcd.init();
  lcd.setRotation(1); 
  lcd.clear(TFT_RED);
  if (!SD.begin(sdSelectPin))
  {
    Serial.println("SD Card initialisation failed!");
  }
  Serial.println("Ready");
}
void loop() {
  
}`

**Reply by sukesh-ak** [reply]
Click `new discussion` from here
https://github.com/sukesh-ak/ESP32-TUX/discussions

**Comment by modi12jin** [comment]
![mmexport45addd7a9e1c25084455942b1c6b66ad_1687254298386](https://github.com/sukesh-ak/ESP32-TUX/assets/40233017/099e0242-1435-488b-9f71-4ba355b77f72)
![mmexporta8777ac6060408d4a0a22e3fb0b813f5_1687254301911](https://github.com/sukesh-ak/ESP32-TUX/assets/40233017/b37bd389-5057-41f3-913f-f4a9163794df)
![mmexportf360cad2ee3b49dcdedfe9c2a1b86948_1687254309088](https://github.com/sukesh-ak/ESP32-TUX/assets/40233017/8c013a99-e1c8-43f8-8a81-ec8787981fbd)

@sukesh-ak ESP32P4 dot screen interface is really complete: SPI, QSPI, 8080, RGB, mipi DSI all have

**Reply by sukesh-ak** [reply]
Thank you for the info. I don't have devices to play with it other than S3. 

Any idea when an HMI device with ESP32-P4 would be available? 
I hope Apache NuttX support would be a priority for it as well.

**Comment by modi12jin** [comment]
![mmexport5040b18cd687ef4d6fe35e8dcd1bd269_1687255105475](https://github.com/sukesh-ak/ESP32-TUX/assets/40233017/8a5d6e87-ab78-4c42-9c52-eceeee70f201)

I’m not too sure, the information I’ll disclose to you is all the content introduced in this seminar

https://mp.weixin.qq.com/s/63by9FsLAaohMSDbgbfQDA

**Reply by sukesh-ak** [reply]
That's nice. I don't access to that link. Maybe it is geo-restricted content.

**Reply by modi12jin** [reply]
@sukesh-ak I don't know if you have seen this warehouse, it is very helpful for us to optimize LVGL, but unfortunately did not introduce the 8080 interface

![hardware_flow](https://github.com/sukesh-ak/ESP32-TUX/assets/40233017/ed24b856-f463-4440-83af-5c2ad1810dd2)

https://github.com/Lzw655/esp32-s3-lcd-ev-baord-docs/tree/master/Development%20guide/en

**Reply by modi12jin** [reply]
https://forms.office.com/Pages/ResponsePage.aspx?id=_SevX1c1lEKVRY6nSkCfOWu_YZd0cddApgv4uGu48n1UOFJWUTBGV0tVT0oxS1gyUTRMUTlFNUczVS4u&origin=QRCode

Can you access this link?

**Comment by modi12jin** [comment]
@sukesh-ak If you can understand Chinese documents, then read Chinese documents.  I found that the author updated the content of the English document lagging behind the Chinese document

https://github.com/Lzw655/esp32-s3-lcd-ev-baord-docs/tree/master/Development%20guide/zh_CN

**Comment by modi12jin** [comment]
@sukesh-ak There is a document in the esp_lvgl_port component, which introduces LCD and LVGL performance

https://github.com/espressif/esp-bsp/blob/master/components/esp_lvgl_port/docs/performance.md

**Reply by sukesh-ak** [reply]
Will check it out. Most drivers are missing and they update only the drivers for Espressif dev kits.

**Reply by modi12jin** [reply]
@sukesh-ak I got official news from Espressif Systems that the ESP32P4mipi DSI is dual channel.  Shanghai is expected to get it by the end of October

**Reply by sukesh-ak** [reply]
Good to know, so faster Displays!😁

Hope they make it work with Apache NuttX as well. 

Btw ESP32-P4 needs external support for RF (WiFi/ BLE).
